projects/data-grid/src/lib/store/data-grid.ts

```diff
@@ -250,6 +244,26 @@ const toggleDetailGrid = (state: GridState, {child, active}: ToggleDetailsGridPa
     }, state)
     : state;
 
+const deleteRow = (state: GridState, action: DeleteRowByIndexPayload | DeleteRowWherePayload<any>): GridState => {
+  const rowIndex = (action as DeleteRowByIndexPayload).rowIndex;
+
+  if (hasValue(rowIndex) && !Number.isNaN(rowIndex)) {
+    const dataItemIndex = state.rowDataIndexes[rowIndex];
+    return dataItemIndex > -1 ? R.evolve({
+      rowDataIndexes: R.remove(rowIndex, 1),
+      data: R.remove(dataItemIndex, 1)
+    }, state) : state;
+  } else if ((action as DeleteRowWherePayload<any>).where) {
+    const dataItemIndex = R.findIndex((action as DeleteRowWherePayload<any>).where, state.data);
+    return dataItemIndex > -1 ? R.evolve({
+      rowDataIndexes: R.remove(dataItemIndex, 1),
+      data: R.remove(dataItemIndex, 1)
+    }, state) : state;
+  }
```

Comment:
In order to avoid code duplication, isn't better first to resolve index of data item and after that to performa the state update.